### PR TITLE
Pda.mcmc updates

### PR DESCRIPTION
**Pull request by ryankelly-uiuc**
Updates to pda.mcmc.R that bring it up to date with the rest of Pecan and allow it to be run with a single function call. Added to both /scripts/workflow.R and /web/workflow.R, though I haven't been able to test the latter. The function immediately checks whether an 'assim.batch' section is in pecan.xml, and aborts if not. So these changes shouldn't interfere with any existing code, but if a user adds the appropriate tags to pecan.xml (or supplies them directly as arguments to pda.mcmc.R) they can do basic brute force data assimilation. 

Current version can only assimilate Ameriflux NEE, but is written with generalization in mind.

### CODE REVIEW

**Comment by mdietze** [comment]
Two things I noted: 

* you shouldn't be inserting records into the inputs database here, you should be querying the database to find existing records

* the calculation of the flux uncertainty heteroskedasticity should not be within the likelihood, but should be precomputed since they don't change with every iteration thought the mcmc (and isn't super cheap)

File: modules/assim.batch/R/pda.mcmc.R [hunk 1]
```diff
@@ -1,40 +1,78 @@
...
+pda.mcmc <- function(settings, params=NULL, jvar=NULL, var.names=NULL, prior=NULL, chain=NULL,
+                     adapt=NULL, adj.min=NULL, ar.target=NULL){
+  # Quit if pda not requested in settings
+  if(!('assim.batch' %in% names(settings))) {
+    return()
+  }
+
+  require(coda)
   
   ## this bit of code is useful for defining the variables passed to this function
```
Comment (dlebauer) [inline modules/assim.batch/R/pda.mcmc.R]: This was one of the toughest things I found about debugging R functions - having the objects you need inside the function in the form that was causing a bug.

A few solutions I've found:

* `debugonce(pda.mcmc)` before running the function will put you inside the function with all of the variables you need. This is the most important all-around workhorse
* but to get there you need an example with such inputs. These can go, e.g.
 * in tests 
 * in vignettes 
 * in function examples

**Comment by ryankelly-uiuc** [comment]
Thanks for the tip on `debugonce()`, @dlebauer . Somehow I had never come across that. I do use `browser()` to similar effect often, though it doesn't play nicely with my text editor's 'run in R' functionality.

**Comment by ryankelly-uiuc** [comment]
@mdietze , per discussion on hipchat:
> you shouldn't be inserting records into the inputs database here, you should be querying the database to find existing records

Left this for now because we need the input to have an id in the database, otherwise bety won't allow th likelihood to be added. Another possibility is to add neither the input or the likelihood, of course.

> the calculation of the flux uncertainty heteroskedasticity should not be within the likelihood, but should be precomputed since they don't change with every iteration thought the mcmc (and isn't super cheap)

Done

**Pull request merged**